test/libsolidity/util/BytesUtils.cpp

```diff
@@ -206,6 +222,31 @@ string BytesUtils::formatString(bytes const& _bytes, size_t _cutOff)
 	return os.str();
 }
 
+std::string BytesUtils::formatFixedPoint(bytes const& _bytes, bool _signed, size_t _fractionalDigits)
+{
+	string value;
+	if (_signed)
+		value = u2s(u256{_bytes}).str();
+	else
+		value = u256{_bytes}.str();
+	bool isNegative = boost::starts_with(value, "-");
+	if (isNegative)
+		boost::erase_all(value, "-");
+	if (_fractionalDigits > 0)
+	{
+		if (_fractionalDigits > value.length())
+		{
+			value.insert(0, string(_fractionalDigits - value.length() + 1, '0'));
+			value.insert(value.length() - _fractionalDigits, ".");
+		}
+		else
+			value.insert(value.length() - _fractionalDigits, ".");
+	}
+	if (isNegative)
+		value.insert(0, "-");
+	return value;
```

Comment:
The string-based sign check looks a bit weird.
```suggestion
	string value;
	bool negative = false;
	if (_signed)
	{
	    s256 signedValue{u2s(u256{_bytes}));
	    if (signedValue < 0)
	    {
	      negative = true;
	      signedValue = -signedValue;
	    }
	    value = signedValue.str();
	}
	else
		value = u256{_bytes}.str();
	if (_fractionalDigits > 0)
	{
		if (_fractionalDigits > value.length())
			value = string(_fractionalDigits - value.length() + 1, '0') + value;
		value.insert(value.length() - _fractionalDigits, ".");
	}
	return (negative ? "-" : "") + value;
```